packages/components/file-input/tests/file-input.test.tsx

```diff
@@ -96,4 +96,193 @@ describe("<FileInput />", () => {
       "true",
     )
   })
+
+  test("should receive the resetRef, which resets files when invoked", () => {
+    const resetRef: { current: (() => void) | null } = { current: null }
+    let files: any = null
+    const setFiles = jest.fn((newFiles: any) => {
+      files = newFiles
+    })
+    const { container } = render(
+      <FileInput
+        data-testid="FileInput"
+        multiple
+        resetRef={resetRef}
+        onChange={setFiles}
+      />,
+    )
+    const mockedFile1 = new File(["foo"], "foo.txt", { type: "text/plain" })
+    const mockedFile2 = new File(["images"], "image.png", { type: "image/png" })
+    const mockedFile3 = new File(["pdf"], "doc.pdf", {
+      type: "application/pdf",
+    })
+    const fileInput = container.querySelector('input[type="file"]')!
+    const mockedFiles = [mockedFile1, mockedFile2, mockedFile3]
+    fireEvent.change(fileInput, {
+      target: { files: mockedFiles },
+    })
+
+    expect(files).toBeTruthy()
+    if (files) {
+      expect(files.length).toEqual(mockedFiles.length)
+      for (let i = 0; i < mockedFiles.length; ++i) {
+        expect(mockedFiles[i]).toEqual(files[i])
+      }
+    }
+    expect(resetRef).toBeTruthy()
+    expect(typeof resetRef.current).toEqual("function")
+    act(() => {
+      resetRef.current && resetRef.current()
+    })
+    expect(files).toBeFalsy()
+    expect(setFiles).toHaveBeenCalledTimes(2)
+  })
+
+  test("should receive the files onChange", () => {
+    let files: any = null
+    const setFiles = jest.fn((newFiles: any) => {
+      files = newFiles
+    })
+
+    const { container } = render(
+      <FileInput data-testid="FileInput" multiple onChange={setFiles} />,
+    )
+
+    const mockedFile = new File(["foo"], "foo.txt", { type: "text/plain" })
+    const fileInput = container.querySelector('input[type="file"]')!
+
+    fireEvent.change(fileInput, {
+      target: { files: [mockedFile, mockedFile, mockedFile] },
+    })
+
+    expect(setFiles).toHaveBeenCalledTimes(1)
+    expect(files).toBeTruthy()
+    if (files) {
+      expect(files.length).toEqual(3)
+      files.forEach((file: File) => expect(file).toEqual(mockedFile))
+    }
+  })
+
+  test("should not not show files in the onChange function if no files are present", () => {
```

Comment:
It might be better to use only one "not".

Reply:
I changed it, sorry for that.